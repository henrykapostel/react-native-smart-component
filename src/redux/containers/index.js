import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { createSelector } from 'reselect'
import { Actions } from './../actions'

export const userInfo$ = state => state.userInfo
const userSelector$ = createSelector(userInfo$, userInfo => ({
  userInfo
}))

export const userFeedPhoto$ = state => state.userFeedPhoto
const userFeedSelector$ = createSelector(userFeedPhoto$, userFeedPhoto => ({
  userFeedPhoto
}))

export const popularFeedPhoto$ = state => state.popularFeedPhoto
const popularFeedSelector$ = createSelector(popularFeedPhoto$, popularFeedPhoto => ({
  popularFeedPhoto
}))

const mapStateToProps = state => ({
  ...userSelector$(state),
  ...userFeedSelector$(state),
  ...popularFeedSelector$(state)
})
const mapDispatchToProps = dispatch =>
  bindActionCreators({ ...Actions }, dispatch)

export default component =>
  connect(mapStateToProps, mapDispatchToProps)(component)
