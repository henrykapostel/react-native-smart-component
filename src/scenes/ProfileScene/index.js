import React, { Component } from 'react'
import { View, ActivityIndicator } from 'react-native'
import { Profile } from '../../component/Profile'
import { getPopularFeedPhoto, getUserFeedPhoto, getUserInfo } from '../../utilities'
import { styles } from './style'

class _ProfileScene extends Component {
  static propTypes = {
  }
  
  static navigationOptions = {
    title : 'Profile',
    header: null,
  };
  
  constructor(props) {
    super(props)
    this.state = {
      isLoading: false
    }
  }
  
  componentDidMount() {
    this.onGetInfo()
  }
  
  onGetInfo = async () => {
    this.setState({ isLoading: true })
    try {
      const userInfo         = await getUserInfo()
      const userFeedPhoto    = await getUserFeedPhoto()
      const popularFeedPhoto = await getPopularFeedPhoto()
      
      this.props.setUserInfo(userInfo)
      this.props.setUserFeed(userFeedPhoto)
      this.props.setPopularFeed(popularFeedPhoto)
      
      this.setState({ isLoading: false })
    } catch (e) {
      this.setState({ isLoading: false })
      console.info('error~~', e)
    }
  }
  
  renderLoading = () => {
    const { isLoading } = this.state
    if (isLoading) {
      return (
        <View style={styles.loadingContainer}>
          <ActivityIndicator size="small" color="white" />
        </View>
      )
    }
  }
  
  renderContent = () => {
    const { isLoading } = this.state
    if (!isLoading) {
      return (
        <Profile {...this.props} />
      )
    }
  }
  
  render() {
    return (
      <View style={styles.container}>
        {this.renderContent()}
        {this.renderLoading()}
      </View>
    )
  }
}

import userContainer from './../../redux/containers'
const ProfileScene = userContainer(_ProfileScene)
export default ProfileScene