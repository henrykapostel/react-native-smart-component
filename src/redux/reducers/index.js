import { combineReducers } from 'redux'
import { userFeedPhoto, popularFeedPhoto, userInfo } from './Reducers'

export default combineReducers({
  userFeedPhoto,
  popularFeedPhoto,
  userInfo
})
