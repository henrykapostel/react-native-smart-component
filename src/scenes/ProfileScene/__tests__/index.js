import 'react-native'
import React from 'react'
import { ProfileScene } from './../../../scenes'
import { store } from './../../../redux/store'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

it('ProfileScene renders correctly', () => {
  const tree = renderer.create(
    <ProfileScene store={store}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})