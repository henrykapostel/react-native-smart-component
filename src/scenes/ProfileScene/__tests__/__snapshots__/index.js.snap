// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileScene renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "gray",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "opacity": 0.5,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="white"
      hidesWhenStopped={true}
      size="small"
    />
  </View>
</View>
`;
